### ISSUE TITLE
Protecting video from downloading

### ISSUE THREAD

**Issue by sigaev-pro** [kylegilman/video-embed-thumbnail-generator#369]
Hi!

I'm building a video blog which can be accessed for registered users only, and I want something, what can prevent my videos from downloading by users. I understand, that anything, what user seen on him screen can be captured and it is no 100% solution, but is it a way to have some protection layer?

**Comment by kylegilman** [comment]
At the moment I don’t have any system for obscuring file URLs. I’m hoping to support token-based authentication for video URLs in the future, but I have a lot to learn about it before I can implement it. It’s also possible to keep the video files in a non-web-accessible folder and stream video playback through PHP, which I believe is what [this plugin](https://wordpress.org/plugins/secure-html5-video-player/) does. But that takes up a lot of system resources and I can’t figure out if it’s actually worth doing. Tokenization seems like a better, more scalable solution. But neither of these systems work with my plugin at the moment.

**Comment by n0risc** [comment]
Simply put htaccess rule to deny .mp4 or others in /uploads directory. But you obviously need to be delivering the videos from a outside media server symlinked.  This prevents the videos being played in the attachment ediyor however.

**Comment by n0risc** [comment]
Kyle has added filter to inject your own playback url with the kgvid_single_code function. There is a plugin someplace that is voded to do this. Called vod-streaming.. look through the issue archives.

**Comment by kylegilman** [comment]
htaccess protection has been known to cause playback problems on mobile devices. Mostly on iPhones because the video player app doesn't get the authorization passed from the browser. I don't know if this is still a problem.

**Comment by n0risc** [comment]
Kyle: Im actually coding a plugin that will effectively incorp your plugin and be able to dynamically generate set manifests and deliver videos from AMS or Wowza. Then blocking /uploads folder with .htaccess will prevent progressive downloading.

**Comment by n0risc** [comment]
However it breaks ediior playback on admin side.

**Comment by benhartwich** [comment]
Hi,
you can use this project, which has implemented wowza streaming with this plugin: https://develop.intelec.uni-passau.de/gitlab/medien/vod-streaming

**Comment by n0risc** [comment]
@benhartwich  Yes, I've extensively added a ton of code to your plugin, or Thomas's..  It actually generates the .m3u8 manifests too on the fly for adaptive playback.  Works well, will reveal it soon on WP & Github.

**Comment by benhartwich** [comment]
OK, would be great, when you could commit your changes to our gitlab repository :)
